test/test_srv_polling.py

```diff
@@ -230,6 +230,47 @@ def erroring_seedlist():
             raise ConfigurationError
         self._test_recover_from_initial(erroring_seedlist)
 
+    def test_10_all_dns_selected(self):
+        response = [("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc", 27019),
+                    ("localhost.test.build.10gen.cc", 27020)]
+
+        def nodelist_callback():
+            return response
+
+        with SrvPollingKnobs(
+                ttl_time=WAIT_TIME, min_srv_rescan_interval=WAIT_TIME,
+                nodelist_callback=nodelist_callback):
+            client = MongoClient(self.CONNECTION_STRING, srvMaxHosts=0)
+            sleep(2*common.MIN_SRV_RESCAN_INTERVAL)
+            expected_topology = {
+                ("localhost.test.build.10gen.cc", 27017),
+                ("localhost.test.build.10gen.cc",  27019),
+                ("localhost.test.build.10gen.cc",  27020)}
+            final_topology = set(
+                client.topology_description.server_descriptions())
+            self.assertEqual(final_topology, final_topology | expected_topology)
+
+    def test_11_all_dns_selected(self):
+        response = [("localhost.test.build.10gen.cc", 27019),
+                    ("localhost.test.build.10gen.cc", 27020)]
+
+        def nodelist_callback():
+            return response
+
+        with SrvPollingKnobs(
+                ttl_time=WAIT_TIME, min_srv_rescan_interval=WAIT_TIME):
+            client = MongoClient(self.CONNECTION_STRING, srvMaxHosts=2)
+            #self.assert_nodelist_change(response, client)
+            with SrvPollingKnobs(nodelist_callback=nodelist_callback):
+                sleep(2*common.MIN_SRV_RESCAN_INTERVAL)
+                expected_topology = {
+                    ("localhost.test.build.10gen.cc",  27019),
+                    ("localhost.test.build.10gen.cc",  27020)}
+                final_topology = set(
+                    client.topology_description.server_descriptions())
+                self.assertSetEqual(final_topology, final_topology | expected_topology)
```

Comment:
Because we don't want to assert that they are equal because the spec never says that they are equal, just that `expected_topology` is a subset of `final_topology`.